src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -62,7 +64,7 @@ def post(url, body, headers, options = {})
     request_options.set_headers(headers)
     request_options.set_as(ResponseBodyType::TEXT)
     request_options.set_body(body)
-    response = @client.post(request_options)
+    response = @@client.post(request_options)
```

Comment:
This could be changed to call a class method, e.g. `self.client_post(request_options)`.

The implementation of that class method might look something like this:

```
def self.client_post(request_options)
   self.client.post(request_options)
end

def self.client()
   @client ||= self.create_client
end
```

and then `self.create_client` would contain the logic that actually constructs the client instance.